src/main/java/fi/aalto/cs/apluscourses/intellij/model/IntelliJModule.java

```diff
@@ -176,4 +182,25 @@ protected boolean hasLocalChanges(@NotNull ZonedDateTime downloadedAt) {
         + PluginSettings.REASONABLE_DELAY_FOR_MODULE_INSTALLATION;
     return ReadAction.compute(() -> VfsUtil.hasDirectoryChanges(fullPath, timeStamp));
   }
+
+  /*
+   * Creates a file with commands to be run on Scala REPL startup (in case commands are listed for
+   * the Module in course configuration file).
+   */
+  // todo: test me!
+  protected void writeReplInitialCommandsFile() {
+    String[] commands = getReplInitialCommandsForModule(
+        Objects.requireNonNull(this.getPlatformObject()));
+    if (commands != null && !ArrayUtils.isEmpty(commands)) {
+
+      File file = new File(getFullPath() + "/"
+          + PluginSettings.MODULE_REPL_INITIAL_COMMANDS_FILE_NAME);
+      try {
+        FileUtils.writeLines(file, StandardCharsets.UTF_8.name(), Arrays.asList(commands));
+      } catch (IOException ex) {
+        logger.error("Could not write REPL initial commands file", ex);
+      }
+      FileUtilRt.createIfNotExists(file);
+    }
+  }
```

Comment:
This should probably `private`.

Does this mean that we only support start-up commands for modules that are downloadable? An alternative is to create the file when the REPL is launched, then it would work for any module. This can be changed later too though

Reply:
this is a good point, however, I think it's good as is because:
* if the module does is not downloaded (e.g. created by hand), then it won't have `replInitialCommands` in our `Module` object (at least with current prospects) - so even if there is a file it still won't have any commands
* if the module like `Ave` is meant to teach students to create a module "from scratch", then they can eventually create a file by hand and populate it with whatever commands/imports they like (and if the file has the right name the commands will be played by the REPL)
* if the intention is to go even more "from scratch", they can learn to use Scala REPL `-i` flag that will allow them to initially execute any script (file with commands)

Reply:
with `protected` I won't have to `PowerMockito` the method if I'll ever get to writing a test for it (which I probably should have done already :rofl: )